### PR TITLE
Use native S3 client

### PR DESCRIPTION
**Pull request by scottcrossen**
This should work better with China because the connection endpoint is chosen based on location of the EC2 node whenever ansible_puller is called from EC2 (this is due mostly to the new version of the SDK)

Asking for a preliminary review before I test it.

The AWS SDK requires a context for all operations so I worked that through the binary.

### CODE REVIEW

File: idempotent_download.go [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+	"errors"
 	"github.com/sirupsen/logrus"
 	"io"
 	"os"
 )
 
 // Interface with logic to govern how to actually pull objects
 type downloader interface {
-	Download(remotePath, outputPath string) error
-	DownloadAndValidateChecksum(remotePath, outputPath, checksum string) error
```
Comment (scottcrossen) [inline idempotent_download.go:14]: I've removed `DownloadAndValidateChecksum` because this logic is the same for both client types

**Comment by scottcrossen** [comment]
I've added an explicit `s3-conn-region` option for users to specify which region to connect to s3 from. Adam Kafka and I have also tested this in cn-north-1

**Review by adkafka — APPROVED** [review]
Looks good to me! Curious about why we added in the `context` stuff though.

**Review by jyates — APPROVED** [review]
lgtm, thanks for digging into this scott!

File: main.go [hunk 2]
```diff
@@ -272,8 +278,10 @@ func ansibleRun() error {
 }
 
 func main() {
+	ctx := context.Background()
```
[the remark this answers is not in the document]
Comment (adkafka) [reply main.go:281]: Makes sense. I think the current implementation is fine. I am no golang expert either.

[the remark this answers is not in the document]
Comment (scottcrossen) [reply main.go]: Well @dfinninger is better than me so I asked his opinion. We decided to use `context.TODO()` for now as we don't really have precedent in this repo yet. If desired, we can come back and cleanup everything to use the same base context.

**Review by dfinninger — APPROVED** [review]
LGTM, thanks @scottcrossen !

**Pull request merged**